code/modules/clothing/accessories/accessory.dm

```diff
@@ -59,6 +61,38 @@
 	inv_overlay = null
 	return ..()
 
+//Defining this at item level to prevent CASTING HELL
+/obj/item/proc/generate_accessory_overlays()
+	return
+
+/obj/item/clothing/generate_accessory_overlays(var/obj/Overlays/O)
+	if(accessories.len)
+		for(var/obj/item/clothing/accessory/accessory in accessories)
+			O.overlays += image("icon" = 'icons/mob/clothing_accessories.dmi', "icon_state" = "[accessory._color || accessory.icon_state]")
```

Comment:
This should be a proc on the accessory itself in case accessories want to modify their mob overlays, and that code should probably include a stored image to reduce the generation overhead.